front-end-tools/src/components/InitComponent.tsx

```diff
@@ -0,0 +1,643 @@
+                                        });
+
+                                        register.onChange(e);
+
+                                        setParsingError(undefined);
+                                    }}
+                                />
+                                {form.formState.errors.inputParameter && (
+                                    <Alert variant="info"> Input parameter is required </Alert>
+                                )}
+                                <Form.Text />
+                            </Form.Group>
+                        )}
+
+                        {(inputParameterType === 'object' || inputParameterType === 'array') && (
+                            <Form.Group className="mb-3">
+                                <Form.Label> Add your input parameter ({inputParameterType}):</Form.Label>
+
+                                {inputParameterType === 'array' && (
+                                    <textarea
+                                        {...form.register('inputParameter')}
+                                        onChange={(event) => {
+                                            setParsingError(undefined);
+                                            const target = event.target as HTMLTextAreaElement;
+
+                                            try {
+                                                JSON.parse(target.value);
+                                            } catch (e) {
+                                                setParsingError((e as Error).message);
+                                                return;
+                                            }
+                                            form.setValue('inputParameter', target.value);
+                                        }}
+                                    >
+                                        {getArrayExample(inputParameterTemplate)}
+                                    </textarea>
+                                )}
+                                {inputParameterType === 'object' && (
+                                    <textarea
+                                        {...form.register('inputParameter')}
+                                        onChange={(event) => {
+                                            setParsingError(undefined);
+                                            const target = event.target as HTMLTextAreaElement;
+
+                                            try {
+                                                JSON.parse(target.value);
+                                            } catch (e) {
+                                                setParsingError((e as Error).message);
+                                                return;
+                                            }
+                                            form.setValue('inputParameter', target.value);
+                                        }}
```

Comment:
I think the correct way to avoid duplication would be to add additional components since this is not the only part that is `duplicated`. I added it as a future task and keep it as it is for now.

```
Refactor: The text area used as input parameter should be its own component to avoid code duplication. The `inputParameterType` dropDown and the `inputParameter` field/textArea should be its own component to avoid code duplication.
```
https://github.com/Concordium/concordium-smart-contract-tools/issues/131

Reply:
Okay. But I would still argue that this should be refactored to use the forms library as intended.